deepchem/models/keras_model.py

```diff
@@ -460,28 +476,49 @@ def fit_generator(self,
 
       if checkpoint_interval > 0 and current_step % checkpoint_interval == checkpoint_interval - 1:
         manager.save()
+        for ext_logger in self.loggers:
+          if isinstance(ext_logger, WandbLogger):
+            ext_logger.save_checkpoint(self.model_dir,
+                                      self,
+                                      "train_checkpoints",
+                                      "step",
+                                      current_step,
+                                      max_checkpoints_to_keep,
+                                      checkpoint_on_min=False)
       for c in callbacks:
         c(self, current_step)
       if self.tensorboard and should_log:
         self._log_scalar_to_tensorboard('loss', batch_loss, current_step)
-      if (self.wandb_logger is not None) and should_log:
-        all_data = dict({'train/loss': batch_loss})
-        self.wandb_logger.log_data(all_data, step=current_step)
+      for ext_logger in self.loggers:
+        if isinstance(ext_logger, WandbLogger):
+          ext_logger.log_batch({"loss": batch_loss}, current_step, inputs, labels, group="train")
+        else:
+          ext_logger.log_batch({"loss": batch_loss}, current_step, inputs, labels)
```

Comment:
Everything should be done through the generic interface.  The whole point of a generic interface is that you don't need special case code for different subclasses.

Reply:
Are you suggesting something like this:
```
for ext_logger in self.loggers:
    ext_logger.log_batch({"loss": batch_loss}, current_step, inputs, labels, group="train")
```
If we don't check the type of the logger wouldn't it be problematic if the `log_batch` function signature differs between loggers? For example, the `log_batch` in WandbLogger adds an extra `group` parameter, but this parameter would just be a Wandb thing and not universal to all Loggers. The core log_batch function would be `def log_batch(data, current_step, inputs, labels)`.

Reply:
All loggers will have identical signatures for all methods.  That's the point of a common interface.  The parent class defines the API, and all subclasses have to use it.

It's ok to include `kwargs` as part of the signature.  But if KerasModel and TorchModel will be passing an argument, it probably ought to be part of the standard API.